### ISSUE TITLE
Electron 4 and vibrancy

### ISSUE THREAD

**Issue by neegool** [sindresorhus/caprine#694]
In reference to #667, release 2.24.0's electron version now detects the system-wise dark mode status. It can be seen in the sidebar. Selecting Caprine's light mode while using system-wide dark mode makes the sidebar look wrong. It's because internally I've set the sidebar vibrancy mode to `sidebar`, which I think changes based on the state of the system dark mode.

With that being said, I plan on making a new pull request which fixes the issue. We have two options here:

1. *Stick with the old behavior*: app-level dark mode can be set independently of the system-level dark mode. This means that I should avoid using `sidebar` vibrancy profile, and revert to using `light`

2. *Adapt to the system-level dark mode*: This means that we will remove the option to set the app-level dark mode.

I'd like to know your thoughts regarding the direction you want the app to move towards.

**Comment by sindresorhus** [comment]
Both.

1. We can use https://electronjs.org/docs/api/system-preferences#systempreferencessetapplevelappearanceappearance-macos to set the native appearance.
2. https://github.com/sindresorhus/caprine/issues/510

**Comment by neegool** [comment]
Got it, so I suppose the new menu option would be something like this?

```
App theme
 - Auto
 - Light
 - Dark
```

**Comment by sindresorhus** [comment]
I considered doing:

```
Theme:
- System
- Light
- Dark
```

But then it's harder to add a toggle shortcut, and the `System` would only be there on macOS.

So maybe this would be better?

```
Follow System Appearance
Dark Mode
```

`Follow System Appearance` would only be shown on macOS. If `Follow System Appearance` is checked, then the `Dark Mode` menu item would be disabled.

**Comment by neegool** [comment]
I think the first one can be represented in the menu via radio buttons, but with the way the config will be structured, I guess the second suggestion is better.

While trying to implement this, I noticed that I cannot access `systemPreferences` when I instantiate it from within `browser.js`. I believe it has to do with context isolation? Accessing it from `index.js` works perfectly. Is there a way I could get `systemPreferences.isDarkMode()` from within `browser.js`?

**Comment by sindresorhus** [comment]
```js
const {api} = require('electorn-util');

api.systemPreferences.isDarkMode();
```

**Comment by neegool** [comment]
Thanks!

I also realized that in `MenuItem`, `enabled` is not data-bound, so it will not update when the boolean variable assigned to it changes, unlike `checked`. Therefore, disabling the `Dark Mode` option dynamically might be too cumbersome to implement.

In this case, I might go back to this menu layout:
```
Theme:
- System
- Light
- Dark
```

**Comment by sindresorhus** [comment]
> I also realized that in MenuItem, enabled is not data-bound, so it will not update when the boolean variable assigned to it changes, unlike checked. Therefore, disabling the Dark Mode option dynamically might be too cumbersome to implement.

The menu is hidden after each click/change, so there's no dynamism needed.

**Comment by neegool** [comment]
That's what I thought as well, but on my end it seems like `enabled` gets its value only once -- during app startup.

**Comment by sindresorhus** [comment]
Ah yes, because the menu is initiated statically. Seems we can do it dynamically though. See the example here: https://github.com/electron/electron/issues/11891
